### PR TITLE
refactor: port server methods to node

### PR DESCRIPTION
**Pull request by agostbiro**
Wraps up https://github.com/NomicFoundation/edr/issues/141

I didn't manage to get to a submittable state yesterday due to difficulties with `hardhat_mine` and decided the combine my changes in the interest of wrapping up the refactor this week. I repeatedly tried to move files with `git mv`, but unfortunately when committing, Git decides to drop the references to the old files, so it treats `crates/edr_rpc_server/src/node/node.rs` and `crates/edr_rpc_server/src/server.rs` as new files. My apologies for making the review difficult.

Regarding `hardhat_mine`: the original Rust implementation didn't seem to follow the logic of the [JavaScript implementation](https://github.com/NomicFoundation/hardhat/blob/f5615062eefb96614981830c12e0636698ae5759/packages/hardhat-core/src/internal/hardhat-network/provider/node.ts#L433) and the logging part was a noop. I tried to reimplement it correctly, but I didn't succeed until now, so I decided to remove it in the interest of submitting a PR today. I created a follow up issue to implement `hardhat_mine` here: https://github.com/NomicFoundation/edr/issues/218. The priority on the follow up issue is unclear to me.

### CODE REVIEW

**Comment by Wodann** [comment]
We work with `node.rs` in the parent instead of `node/mod.rs` in the child to be explicit with naming, so I moved that file back. This also resolved the place where you allowed the clippy warning and avoids the file being deleted and added.

File: crates/edr_rpc_server/src/node/node_data.rs [hunk 1]
```diff
@@ -0,0 +1,424 @@
+use std::sync::Arc;
+use std::time::{SystemTime, UNIX_EPOCH};
```
Comment (Wodann) [inline crates/edr_rpc_server/src/node/node_data.rs:2]: Merge imports

File: crates/edr_rpc_server/src/node/node_data.rs [hunk 2]
```diff
@@ -0,0 +1,424 @@
...
+};
+use edr_evm::blockchain::{Blockchain, ForkedBlockchain, LocalBlockchain};
+use edr_evm::state::{AccountTrie, TrieState};
+use edr_evm::{
+    blockchain::{BlockchainError, SyncBlockchain},
+    mine_block,
+    state::{IrregularState, StateError, SyncState},
+    AccountInfo, Block, CfgEnv, HashMap, HashSet, MemPool, MineBlockResult, MineOrdering,
+    RandomHashGenerator, SyncBlock, KECCAK_EMPTY,
+};
```
Comment (Wodann) [inline crates/edr_rpc_server/src/node/node_data.rs:20]: Merge imports

File: crates/edr_rpc_server/src/server.rs [hunk 3]
```diff
@@ -0,0 +1,696 @@
+use std::net::{SocketAddr, TcpListener};
+use std::sync::Arc;
```
Comment (Wodann) [inline crates/edr_rpc_server/src/server.rs:2]: Merge imports

File: crates/edr_rpc_server/src/filter.rs [hunk 4]
```diff
@@ -1,82 +1,40 @@
...
-};
-use edr_evm::{
-    blockchain::{BlockchainError, SyncBlockchain},
-    state::StateError,
-};
+use edr_eth::remote::filter::{FilteredEvents, LogOutput};
 
-use super::{_block_number_from_block_spec, _block_number_from_hash};
-
-pub struct _FilterCriteria {
```
Comment (Wodann) [inline crates/edr_rpc_server/src/filter.rs:18]: What happened to the filter criteria and the places it was used?

**Review by Wodann — CHANGES_REQUESTED** [review]

File: crates/edr_rpc_server/src/filter.rs (hunk 4 above)
Comment (agostbiro) [reply crates/edr_rpc_server/src/filter.rs:18]: It wasn't used before (hence the underscore prefix I think)

File: crates/edr_rpc_server/src/node/node_data.rs (hunk 2 above)
Comment (agostbiro) [reply crates/edr_rpc_server/src/node/node_data.rs]: Done in [Merge imports](https://github.com/NomicFoundation/hardhat/pull/4539/commits/0e0a2f349ef63b628abbab1547476d3a621afd6d)

File: crates/edr_rpc_server/src/server.rs (hunk 3 above)
Comment (agostbiro) [reply crates/edr_rpc_server/src/server.rs]: Done in [Merge imports](https://github.com/NomicFoundation/hardhat/pull/4539/commits/0e0a2f349ef63b628abbab1547476d3a621afd6d)

**Comment by agostbiro** [comment]
I fixed the merge conflicts

File: crates/edr_rpc_server/src/node/node_data.rs (hunk 1 above)
Comment (agostbiro) [reply crates/edr_rpc_server/src/node/node_data.rs]: Done in [Merge imports](https://github.com/NomicFoundation/hardhat/pull/4539/commits/0e0a2f349ef63b628abbab1547476d3a621afd6d)

File: crates/edr_rpc_server/src/filter.rs (hunk 4 above)
Comment (Wodann) [reply crates/edr_rpc_server/src/filter.rs:18]: It was in the `struct Filter`:
image

**Review by Wodann — APPROVED** [review]
Approving this to expedite resolution of merge conflicts for Pato

File: crates/edr_rpc_server/src/filter.rs (hunk 4 above)
Comment (Wodann) [reply crates/edr_rpc_server/src/filter.rs:18]: If I'm not mistaken, those get used by the functions in: https://github.com/NomicFoundation/edr/issues/114.

Comment (agostbiro) [reply crates/edr_rpc_server/src/filter.rs:18]: Ah sorry I thought you asked if functionality was lost

**Pull request merged**